### PR TITLE
Add trivia for enums and typedefs

### PR DESCRIPTION
**Pull request by saschanaz**
Exposes `trivia` field for enum values.

### CODE REVIEW

File: lib/webidl2.js [hunk 1]
```diff
@@ -1039,44 +1039,57 @@
...
       const vals = [];
       const ret = current = {
         type: "enum",
         name: sanitize_name(name.value, "enum"),
-        values: vals
+        values: vals,
+        trivia
       };
-      consume("{") || error("No curly for enum");
+      const open = consume("{") || error("No curly for enum");
```
Comment (marcoscaceres) [inline lib/webidl2.js]: “Curly” might be confusing. Maybe “no opening `{`...”

File: lib/webidl2.js [hunk 2]
```diff
@@ -1039,44 +1039,57 @@
...
+      trivia.open = open.trivia;
       let value_expected = true;
       while (true) {
-        if (consume("}")) {
+        const close = consume("}");
```
Comment (marcoscaceres) [inline lib/webidl2.js]: Had expected to see “No closing } for enum” somewhere as an error. Is that possible?

Comment (saschanaz) [reply lib/webidl2.js]: Currently `enum X {` throws with *Unexpected value in enum*, which is weird here. I introduced `eof` type in #185, maybe we can do better with it by `Unexpected end of file`.

Comment (saschanaz) [reply lib/webidl2.js]: Filed #208, we may revisit it later.

**Pull request merged**